src/qibolab/compilers/compiler.py

```diff
@@ -149,17 +151,12 @@ def qubit_clock(el: QubitId):
 
         # process circuit gates
         for moment in circuit.queue.moments:
-            for gate in set(filter(lambda x: x is not None, moment)):
+            for gate in {x for x in moment if x is not None}:
                 delay_sequence = PulseSequence()
                 gate_sequence = self.get_sequence(gate, platform)
-                increment = defaultdict(int)
-                start = max(
-                    (
-                        qubit_clock(el)
-                        for el in {ch_to_qb[ch] for ch in gate_sequence.channels}
-                    ),
-                    default=0.0,
-                )
+                increment = defaultdict(float)
+                active_qubits = {ch_to_qb[ch] for ch in gate_sequence.channels}
```

Comment:
Is `active_qubits` the ones that we are playing pulses on as part of `gate`? Should qubits that only participate in virtual-Zs be in `active_qubits`? Currently they are, I am not sure if they should (just trying to understand the issue mentioned above, but maybe I am wrong).

Reply:
I will check, but the role of `active` qubits (the one I had in mind) is exactly its definition: the qubits where something is happening on their channels.

The problem I was trying to catch regarded those qubits that are involved in the interaction, but have none of its channels involved (because their clock has to be advanced, but this doesn't happen through channels).

I'm going to check the issue.